client/Containers/GuestCheckout.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { Component } from 'react';
+import { Link } from 'react-router-dom';
+import { FormGroup, Label, Input } from '../components/styles/Forms';
+import FeaturedButton from '../components/styles/FeaturedButton';
+import { connect } from 'react-redux';
+import { resetAuth } from '../store/auth/auth';
+import { fetchToken, resetToken } from '../store/auth/token';
+import axios from 'axios';
+
+const mapDispatchToProps = (dispatch) => ({
+  fetchToken: (credentials) => dispatch(fetchToken(credentials)),
+  resetAuth: () => dispatch(resetAuth()),
+  resetToken: () => dispatch(resetToken()),
+});
+
+class GuestCheckout extends Component {
+  state = {
+    firstName: '',
+    lastName: '',
+    email: '',
+    phoneNumber: '',
+  };
+
+  onChange = ({ target: { name, value } }) => {
+    this.setState({ [name]: value });
+  };
+
+  onSubmit = async (event) => {
+    event.preventDefault();
+    const { firstName, lastName, email, phoneNumber } = this.state;
+    const token = window.localStorage.getItem('token');
+    try {
+      await axios.post(
+        '/api/users',
+        { firstName, lastName, email, phoneNumber, password: 'guest' },
+        { headers: { authorization: token } }
+      );
+    } catch (err) {
+      console.error(err);
+    }
```

Comment:
You can probable leave out password here on line 35, as the backend route has handling for when the password is undefined